src/main/java/com/example/demo/services/StudentService.java

```diff
@@ -37,7 +40,11 @@ public boolean updateStudentData(long studentId, Student student) {
         return studentRepository.updateStudentData(studentId, student);
     }
 
-    public boolean deleteStudent(long studentId) {
-        return studentRepository.deleteStudent(studentId);
+    public boolean deleteStudentAndHisGrades(long studentId) {
+        boolean ifDeleted = studentRepository.deleteStudent(studentId);
+        if(ifDeleted){
+            gradeService.deleteStudentGrades(studentId);
+        }
+        return ifDeleted;
```

Comment:
CTRL ALT L -> przeformatuje cały plik. Prawdopodobnie usuniesz to `ifDeleted` bo nie jest to teraz potrzebne